com.unity.sg2/Editor/GraphDeltaRegistry/Registry/GraphType.cs

```diff
@@ -337,7 +391,13 @@ public bool CanConvert(FieldHandler src, FieldHandler dst)
             var dstHgt = GraphTypeHelpers.GetHeight(dst);
             var dstLen = GraphTypeHelpers.GetLength(dst);
 
-            return srcHgt == dstHgt || srcHgt == GraphType.Height.One && srcLen >= dstLen;
+
+            return
+                srcHgt == dstHgt ||
+                srcHgt >= dstHgt && srcLen >= dstLen ||
+                srcHgt == GraphType.Height.One && srcLen >= dstLen ||
+                srcHgt == GraphType.Height.One && srcLen == GraphType.Length.One;
```

Comment:
Realistically, only scalar promotion `srcHgt == GraphType.Height.One && srcLen == GraphType.Length.One` and truncation `srcHgt >= dstHgt && srcLen >= dstLen` should be supported. These are the two rules for conversion acceptance, but because of issues with dynamics, I'm leaving the old conditions for now.